lib/hawkular/base_client.rb

```diff
@@ -175,6 +178,16 @@ def tenant_header
       headers
     end
 
+    # @!visibility private
+    def connect_error(fault)
+      if fault.is_a?(SocketError)
+        HawkularConnectionException.new(fault.to_s)
+      elsif [Errno::ECONNREFUSED, Errno::ECONNRESET, Errno::EHOSTUNREACH,
+             Errno::ENETDOWN, Errno::ENETUNREACH, Errno::ETIMEDOUT].include?(fault.class)
+        HawkularConnectionException.new(fault.to_s, fault.class::Errno)
```

Comment:
another thing that concerns me is that the docs say the names of the errors vary per operation system
http://ruby-doc.org/core-2.2.0/Errno.html
so I'd like to propose to reconsider again - do we need such specific handling for connection errors? this definitely doesn't sound unique to hawkular. 
cc @pilhuhn